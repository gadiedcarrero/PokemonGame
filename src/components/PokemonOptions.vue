<template>
  <div class="options-container">
    <ul class="optionsgroup">
      <li
        :class="{
          'border-red': pokemon.id === selectedAnswer && pokemon.id !== correctAnswer,
          'border-green': pokemon.id === correctAnswer,
        }"
        v-for="pokemon in pokemons"
        :key="pokemon.id"
        @click="$emit('selectionPokemon', pokemon.id)"
      >
        {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: "PokemonOptions",
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
    correctAnswer: {
      type: Number,
    },
    selectedAnswer: {
      type: Number,
    },
  },
  methods: {},
};
</script>

<style scoped>
.border-red {
  position: relative;
}

.border-red::after {
  content: '';
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  left: -1px;
  width: 100%;
  height:100%;
  border-radius: 5px;
  border: 2px solid red;
}

.border-green {
  position: relative;
}

.border-green::after {
  content: '';
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  left: -1px;
  width: 100%;
  height: 100%;
  border-radius: 5px;
  border: 2px solid green;
}

/* Pokemon Options */
ul {
  list-style-type: none;
  display: inline-flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin: 0;
  padding: 0;
  unicode-bidi: normal;
  margin-top: 20px;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>
