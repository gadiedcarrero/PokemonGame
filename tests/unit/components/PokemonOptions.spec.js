import PokemonOptions from "@/components/PokemonOptions";
import { shallowMount } from "@vue/test-utils";
import { pokemons } from "../mocks/pokemons.mock";

describe("PokemonOptions component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        pokemons,
        correctAnswer: 1,
        selectedAnswer: 2
      },
    });
  });

  test("debe de hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe mostrar las 4 opciones", () => {
    const liTags = wrapper.findAll("li");
    expect(liTags.length).toBe(4);
    expect(liTags[0].text()).toBe(pokemons[0].name);
    expect(liTags[1].text()).toBe(pokemons[1].name);
    expect(liTags[2].text()).toBe(pokemons[2].name);
    expect(liTags[3].text()).toBe(pokemons[3].name);
  });

  test('debe emitir "selectionPokemon" con sus respectivos parametros al hacer click', () => {
    const [li1, li2, li3, li4] = wrapper.findAll("li");
    li1.trigger("click");
    li2.trigger("click");
    li3.trigger("click");
    li4.trigger("click");
    
    expect(wrapper.emitted("selectionPokemon").length).toBe(4);
    expect(wrapper.emitted("selectionPokemon")).toEqual([
      [pokemons[0].id],
      [pokemons[1].id],
      [pokemons[2].id],
      [pokemons[3].id],
    ]);
  });

  test("debe comprobar que el primer li tenga la clase border-green", () => {
    const [li1, li2, li3, li4] = wrapper.findAll("li");
    expect(li1.classes("border-green")).toBeTruthy();
    expect(li2.classes("border-red")).toBeTruthy();
  });
});
