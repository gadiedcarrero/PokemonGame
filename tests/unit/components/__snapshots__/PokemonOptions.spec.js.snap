// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions component debe de hacer match con el snapshot 1`] = `
<div class="options-container">
  <ul class="optionsgroup">
    <li class="border-green">bulbasaur</li>
    <li class="border-red">ivysaur</li>
    <li class="">venusaur</li>
    <li class="">charmander</li>
  </ul>
</div>
`;
